src/web/topic/components/TopicPane/TopicPane.tsx

```diff
@@ -1,86 +1,96 @@
-import { AutoStories, ChevronLeft, KeyboardArrowDown } from "@mui/icons-material";
+import {
+  AutoStories,
+  ChevronLeft,
+  ChevronRight,
+  KeyboardArrowDown,
+  KeyboardArrowUp,
+} from "@mui/icons-material";
 import { TabContext, TabList, TabPanel } from "@mui/lab";
 import { Tab } from "@mui/material";
+import { useEffect, useState } from "react";
 
+import { emitter } from "@/web/common/event";
 import { GraphPartDetails } from "@/web/topic/components/TopicPane/GraphPartDetails";
 import { TopicDetails } from "@/web/topic/components/TopicPane/TopicDetails";
 import {
+  Anchor,
   PositionedDiv,
   StyledDrawer,
   TogglePaneButton,
 } from "@/web/topic/components/TopicPane/TopicPane.styles";
 import { TopicViews } from "@/web/topic/components/TopicPane/TopicViews";
-import {
-  setIsTopicPaneOpen,
-  setSelectedTab,
-  usePaneStore,
-} from "@/web/topic/components/TopicPane/paneStore";
 import { useSelectedGraphPart } from "@/web/view/currentViewStore/store";
 
+const ANCHOR_ICON_MAP = {
+  top: KeyboardArrowUp,
+  left: ChevronLeft,
+  bottom: KeyboardArrowDown,
+  right: ChevronRight,
+} as const;
+
+type TopicTabName = "Details" | "Views";
 interface Props {
-  isLandscape: boolean;
+  anchor: Anchor;
+  tabs: TopicTabName[];
```

Comment:
```suggestion
  tabs: [TopicTabName, ...TopicTabName[]];
```

You can choose to use this or not, I can't tell if this is too jank, but this guarantees that our array has at least one tab in it, so we don't need to do `tabs[0] ?? ...` or `if tabs.length === 0` checks below.

Reply:
I did not think about recursive typing. Could do it, but i don't think it prevents the TopicPane from being called with an empty array?